app/helpers/heatmap_helper.rb

```diff
@@ -192,21 +194,27 @@ def self.fetch_top_taxons(
   )
     categories_map = ReportHelper::CATEGORIES_TAXID_BY_NAME
     categories_clause = ""
-    if categories.present?
-      categories_clause = " AND superkingdom_taxid IN (#{categories.map { |category| categories_map[category] }.compact.join(',')})"
-    elsif include_phage
-      categories_clause = " AND superkingdom_taxid = #{categories_map['Viruses']}"
+    unless client_filtering_enabled
```

Comment:
Maybe initialize the clause strings above and use just one if?

Reply:
Add add a comment, why we are bypassing all the filters... (specially since we should probably have changed the request on client)

Reply:
Condensed the `if` statements and added comment 😄